controllers/sms-gateway.js

```diff
@@ -30,73 +32,67 @@ function saveToDb(message, callback) {
+function mapStateFields(update) {
+  const result = {
+    messageId: update.id
+  };
+  result.state = STATUS_MAP[update.status];
+  if (result.state) {
     if(update.reason) {
-      details = { reason: update.reason };
+      result.details = { reason: update.reason };
     }
   } else {
-    newState = 'unrecognised';
-    details = { gateway_status: update.status };
+    result.state = 'unrecognised';
+    result.details = { gateway_status: update.status };
   }
-  updateState(update.id, newState, details, callback);
-}
 
-function updateState(messageId, newState, details, callback) {
-  var updateBody = {
-    state: newState,
-  };
-  if(details) {
-    updateBody.details = details;
-  }
-  messageUtils.updateMessage(messageId, updateBody, callback);
+  return result;
 }
 
 function markMessagesForwarded(messages, callback) {
-  async.eachSeries(
-    messages,
-    function(message, callback) {
-      updateState(message.id, 'forwarded-to-gateway', null, callback);
-    },
-    function(err) {
-      if (err) {
-        // No error throwing here, because no big deal : 'forwarded-to-gateway' status
-        // wasn't saved to DB, so message will stay in 'pending' status and be retried next time.
-        warn(err);
-      }
-      return callback();
-    }
-  );
+  const taskStateChanges = messages.map((message) => { return {
+    messageId: message.id,
+    state: 'forwarded-to-gateway'
+  };});
+
+  messageUtils.updateMessageTaskStates(taskStateChanges, callback);
 }
 
 function getOutgoing(callback) {
   messageUtils.getMessages({ states: ['pending', 'forwarded-to-gateway'] },
-    function(err, pendingMessages) {
-    if (err) {
-      warn(err);
-      return callback(null, []);
-    }
-    var messages = pendingMessages.map(function(message) {
-      return {
-        id: message.id,
-        to: message.to,
-        content: message.message,
-      };
-    });
-    markMessagesForwarded(messages, function() {
-      callback(null, messages);
-    });
+    (err, pendingMessages) => {
+      // TODO: I don't get this. Why wouldn't you return an error here? What
+      //       is the intention behind hiding it from the caller?
+      if (err) {
+        warn(err);
+        return callback(null, []);
+      }
+
+      const messages = pendingMessages.map(message => {
+        return {
+          id: message.id,
+          to: message.to,
+          content: message.message,
+        };
+      });
+
+      markMessagesForwarded(messages, (err) => {
+        if (err) {
+          callback(err);
+        } else {
+          callback(null, messages);
+        }
+      });
   });
 }
 
 // Process webapp-terminating messages
-function processMessages(req, callback) {
+function addNewMessages(req, callback) {
   if (!req.body.messages) {
     return callback();
   }
-  async.eachSeries(req.body.messages, saveToDb, function(err) {
+
+  async.eachSeries(req.body.messages, saveToDb, err => {
     if (err) {
       warn(err);
     }
```

Comment:
As above, we probably want to `callback(err)` now.  Happy to consider it for a different ticket if you prefer.